pkg/distributor/ha_tracker.go

```diff
@@ -226,39 +267,11 @@ func (h *haTracker) loop(ctx context.Context) error {
 	// The KVStore config we gave when creating h should have contained a prefix,
 	// which would have given us a prefixed KVStore client. So, we can pass empty string here.
 	h.client.WatchPrefix(ctx, "", func(key string, value interface{}) bool {
-		replica := value.(*ReplicaDesc)
-		segments := strings.SplitN(key, "/", 2)
-
-		// Valid key would look like cluster/replica, and a key without a / such as `ring` would be invalid.
-		if len(segments) != 2 {
-			return true
-		}
-
-		user := segments[0]
-		cluster := segments[1]
-
-		if replica.DeletedAt > 0 {
-			h.electedReplicaChanges.DeleteLabelValues(user, cluster)
-			h.electedReplicaTimestamp.DeleteLabelValues(user, cluster)
-
-			h.electedLock.Lock()
-			defer h.electedLock.Unlock()
-			userClusters := h.clusters[user]
-			if userClusters != nil {
-				delete(userClusters, cluster)
-				if len(userClusters) == 0 {
-					delete(h.clusters, user)
-				}
-			}
-			return true
+		replica, ok := value.(*ReplicaDesc)
+		if !ok {
+			return false
 		}
```

Comment:
This was added more so as a sanity check to make sure if its not the correct type to return false. Do we agree this is something we should keep in `WatchPrefix`? For HA Tracker.